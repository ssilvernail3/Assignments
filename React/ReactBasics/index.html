<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Shiba Inu</title>
</head>

<body>
    <h1>SHIBAS!!</h1>
    <div id="root"></div>
    <script src="https://unpkg.com/react/umd/react.development.js"></script>
    <script src="https://unpkg.com/react-dom/umd/react-dom.development.js"></script>
    <script src="https://unpkg.com/babel-standalone"></script>
    
    <script src="Shiba.js" type="text/jsx"></script>
    <script src="Sharpei.js" type="text/jsx"></script>
    <script src="RandonNum.js" type="text/jsx"></script>
    <script src="Animal.js" type="text/jsx"></script>
    <script src="App.js" type="text/jsx"></script>
</body>

</html>