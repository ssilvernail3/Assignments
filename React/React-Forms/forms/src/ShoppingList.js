import React, { useState } from "react";
import Item from "./Item";
import NewItemForm from "./NewItemForm";
import { v4 as uuidv4 } from "uuid"; 

const ShoppingList = () => {
    const INITIAL_STATE = [
        {id: uuidv4(), name: 'Peanut Butter', qty: 2},
        {id: uuidv4(), name: 'Oat Milk', qty: 1}
    ]
    const [items, setItems] = useState(INITIAL_STATE)

    const addItem = (name, qty) => {
        setItems(items => [...items, { id: uuidv4(), name, qty }]);
    }
    return (
        <div>
            <h3>Shopping List</h3>
            <NewItemForm addItem={addItem}  />
            <div>
                {items.map(({ id, name, qty }) => <Item id={id} name={name} qty={qty} key={id}/>)}
            </div>
        </div>
    )
}

export default ShoppingList; 
