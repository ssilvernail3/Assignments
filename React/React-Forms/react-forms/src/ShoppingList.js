import React, { useState } from "react";
import Item from "./Item";
import NewItemForm from "./NewItemForm";
import { uuid } from "uuidv4";

const ShoppingList = () => {
    const INITIAL_STATE = [
        {id: 1, name: 'Peanut Butter', qty: 2},
        {id: 2, name: 'Oat Milk', qty: 1}
    ]
    const [items, setItems] = useState(INITIAL_STATE)

    const addItem = (name, qty) => {
        setItems(items => [...items, { name, qty }]);
    }
    return (
        <div>
            <h3>Shopping List</h3>
            <NewItemForm addItem={addItem}  />
            <div>
                {items.map(({ id, name, qty }) => <Item id={id} name={name} qty={qty} key={id}/>)}
            </div>
        </div>
    )
}

export default ShoppingList; 
