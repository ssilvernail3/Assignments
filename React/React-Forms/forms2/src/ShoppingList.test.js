import React from "react";
import { render, fireEvent } from "@testing-library/react";
import ShoppingList from "./ShoppingList";

it('renders without crashing', function () {
    render(<ShoppingList />)
});

it('matches snapshot', function () {
    const { asFragment } = render(<ShoppingList />)
    expect(asFragment()).toMatchSnapshot();
}); 

it('should add new item', function () {
    const { queryByText, getByLabelText } = render(<ShoppingList />);
    const product = getByLabelText("Product");
    const btn = queryByText('Add Item');
    const qty = getByLabelText("Quantity");
    expect(queryByText('Product Name: Chocolate Milk')).not.toBeInTheDocument();
    fireEvent.change(product, { target: { value: 'Chocolate Milk' } });
    fireEvent.change(qty, {target : {value : 5}} )
    fireEvent.click(btn);
    expect(queryByText('Product Name: Chocolate Milk')).toBeInTheDocument();
    expect(queryByText('Quantity: 5')).toBeInTheDocument();


});