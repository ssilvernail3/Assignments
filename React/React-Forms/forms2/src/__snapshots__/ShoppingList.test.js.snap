// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Shopping List
    </h3>
    <form>
      <label
        for="name"
      >
        Product
      </label>
      <input
        id="name"
        name="name"
        placeholder="product name"
        type="text"
        value=""
      />
      <label
        for="qty"
      >
        Quantity
      </label>
      <input
        id="qty"
        max="10"
        min="1"
        name="qty"
        type="range"
        value=""
      />
      <button>
        Add Item
      </button>
    </form>
    <div>
      <div>
        <ul>
          <li>
            Product Name: Peanut Butter
          </li>
          <li>
            Quantity: 2
          </li>
        </ul>
      </div>
      <div>
        <ul>
          <li>
            Product Name: Oat Milk
          </li>
          <li>
            Quantity: 1
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
